src/main/java/net/dv8tion/jda/core/requests/WebSocketClient.java

```diff
@@ -938,7 +938,7 @@ protected void handleEvent(JSONObject raw)
 //        }
 
         JSONObject content = raw.getJSONObject("d");
```

Comment:
Maybe move this into the try catch block below? I just got a JSONException on 286 from this line, missing a print of the faulty JSON.

```
[2017-10-18 12:30:31,889] [ DEBUG] [JDA RateLimit-Queue Pool - Thread 2] n.d.jda.core.requests.Requester: Received response with following cf-rays: [3afad2a07e74646f-FRA]
[2017-10-18 12:30:47,770] [ ERROR] [JDA [1 / 2] MainWS-ReadThread] n.d.j.core.requests.WebSocketClient: Encountered an Exception 
org.json.JSONException: JSONObject["d"] is not a JSONObject.
        at org.json.JSONObject.getJSONObject(JSONObject.java:641)
        at net.dv8tion.jda.core.requests.WebSocketClient.handleEvent(WebSocketClient.java:947)
        at net.dv8tion.jda.core.requests.WebSocketClient.onTextMessage(WebSocketClient.java:666)
        at com.neovisionaries.ws.client.ListenerManager.callOnTextMessage(ListenerManager.java:352)
        at com.neovisionaries.ws.client.ReadingThread.callOnTextMessage(ReadingThread.java:260)
        at com.neovisionaries.ws.client.ReadingThread.callOnTextMessage(ReadingThread.java:238)
        at com.neovisionaries.ws.client.ReadingThread.handleTextFrame(ReadingThread.java:963)
        at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:746)
        at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108)
        at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64)
        at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45)
[2017-10-18 12:31:21,579] [ DEBUG] [fredboat.FredBoat.main()] n.d.jda.core.requests.Requester: Received response with following cf-rays: [3afad3d719096361-FRA]
```

Reply:
Not sure if this is the right PR to do that in

Reply:
Seing as this is already fixed in another pr, not gonna implement here

Reply:
That cannot be moved into the try/catch block because the json is used within the catch

Reply:
Aight. My intent on suggesting that was less concerned with the current bug, more of a general idea to improve future logging.
And Minn is correct, the change is more complicated than moving the two lines into the try catch, I'm sure whoever decides to add this in here or in another PR will figure out something appropriate. Maybe look at other similar places for possible uncaught JSONExceptions and have PR for that after this is merged.